fixtures/dev-env/tests/index.test.ts

```diff
@@ -283,6 +288,35 @@ describe("startDevWorker: ProxyController", () => {
 		});
 	});
 
+	test("InspectorProxyWorker rejects unauthorised requests", async () => {
+		const run = await fakeStartUserWorker({
+			script: `
+				export default {
+					fetch() {
+						return new Response();
+					}
+				}
+			`,
+		});
+
+		// Check validates `Host` header
+		ws = new WebSocket(
+			`ws://${run.inspectorProxyWorkerUrl.host}/core:user:${run.config.name}`,
+			{ setHost: false, headers: { Host: "example.com" } }
+		);
+		let openPromise = events.once(ws, "open");
+		await expect(openPromise).rejects.toThrow("Unexpected server response");
+
+		// Check validates `Origin` header
+		ws = new WebSocket(
+			`ws://${run.inspectorProxyWorkerUrl.host}/core:user:${run.config.name}`,
+			{ origin: "https://example.com" }
```

Comment:
The line above uses `headers: {` whereas you've set origin at the top-level – is that correct?